test/stream/bridge/keboola/keboola_test.go

```diff
@@ -302,6 +305,242 @@ func TestKeboolaBridgeWorkflow(t *testing.T) {
 	ts.checkKeboolaTable(t, ctx, 1, 129)
 }
 
+func TestNetworkIssuesKeboolaBridgeWorkflow(t *testing.T) { // nolint: paralleltest
+	metrics := toxiproxy.NewMetricsContainer(nil)
+	server := toxiproxy.NewServer(metrics, zerolog.New(os.Stderr))
+	go server.Listen("localhost:8474")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	// Update configuration to make the cluster testable
+	configFn := func(cfg *config.Config) {
+		// Enable metadata cleanup for removing storage jobs
+		cfg.Storage.MetadataCleanup.Enabled = true
+		// Disable unrelated workers
+		cfg.Storage.DiskCleanup.Enabled = false
+		cfg.API.Task.CleanupEnabled = false
+
+		// Use deterministic load balancer
+		cfg.Storage.Level.Local.Writer.Network.PipelineBalancer = network.RoundRobinBalancerType
+
+		// In the test, we trigger the slice upload via the records count, the other values are intentionally high.
+		cfg.Storage.Level.Staging.Upload = stagingConfig.UploadConfig{
+			MinInterval: duration.From(1 * time.Second), // minimum
+			Trigger: stagingConfig.UploadTrigger{
+				Count:    10,
+				Size:     1000 * datasize.MB,
+				Interval: duration.From(30 * time.Minute),
+			},
+		}
+
+		// In the test, we trigger the file import only when sink limit is not reached.
+		cfg.Sink.Table.Keboola.JobLimit = 1
+
+		// In the test, we trigger the file import via the records count, the other values are intentionally high.
+		cfg.Storage.Level.Target.Import = targetConfig.ImportConfig{
+			MinInterval: duration.From(30 * time.Second), // minimum
+			Trigger: targetConfig.ImportTrigger{
+				Count:       30,
+				Size:        1000 * datasize.MB,
+				Interval:    duration.From(30 * time.Minute),
+				SlicesCount: 100,
+				Expiration:  duration.From(30 * time.Minute),
+			},
+		}
+
+		// Cleanup should be perfomed more frequently to remove already finished storage jobs
+		cfg.Storage.MetadataCleanup.Interval = 10 * time.Second
+	}
+
+	ts := setup(t)
+	client := toxiproxyClient.NewClient("localhost:8474")
+	proxy, err := client.CreateProxy("source1", ts.sourceURL1[7:len(ts.sourceURL1)-1], ts.sourceURL1[7:])
```

Comment:
If the source url is `localhost:12345` i want to create new proxy on host:port `localhost:1234`.
It can be adjusted to randomized port